### DISCUSSION TITLE
Feature Request: Postgres support

### DISCUSSION THREAD

**Discussion by Radiergummi** [envkey/envkey#101] [Ideas and feedback]
Are there any plans to support additional DBMS for the self hosted version? I'm really in need of something exactly like envkey, but our entire internal infrastructure is built around Postgres. I don't really see myself setting up a dedicated MySQL instance just for envkey, and I'm sure others are in the same spot.  
I think you're using knex already, but I don't know the codebase; would it be a lot of work to support other DBMS? Something like sqlite would also very convenient for small or encapsulated deployments.

**Comment by danenania** [comment]
Hi @Radiergummi, this is possible but it would take some doing as there is some MySQL-specific syntax in a few places and it would also take careful testing to be sure that there are no issues in terms of transactions/isolation/consistency (almost every action in EnvKey is transactional).

We could potentially do this for an enterprise customer, but otherwise I don't see us prioritizing it in the near term. That said, I'd be happy to offer support if a community member wanted to take it on.

**Comment by Radiergummi** [comment]
Appreciate the straightforward reply! I'm going to check our options, but maybe this issue could be used to gauge interest by others?

**Comment by danenania** [comment]
@Radiergummi sure thing, I'll move it to a discussion for that purpose.
